pinot-core/src/main/java/org/apache/pinot/core/minion/rollup/MergeRollupSegmentConverter.java

```diff
@@ -111,9 +112,10 @@ private MergeRollupSegmentConverter(@Nonnull List<File> inputIndexDirs, @Nonnull
     for (DateTimeFieldSpec dateTimeFieldSpec : schema.getDateTimeFieldSpecs()) {
       groupByColumns.add(dateTimeFieldSpec.getName());
     }
-    String timeColumn = schema.getTimeColumnName();
-    if (timeColumn != null) {
-      groupByColumns.add(timeColumn);
+    // TODO: once time column starts showing up as dateTimeFieldSpec (https://github.com/apache/incubator-pinot/issues/2756) below lines becomes redundant
```

Comment:
Isnt it safer to keep the API to get time column name, but deprecate it. Do this change when the you actually include the datetimefieldspecs to return the time col as well. 

Otherwise it may pose a danger that the time column appears twice in the groupByColumns List.

Or, turn it into a set, and we are safe.